src/main/java/seedu/taskell/logic/parser/Parser.java

```diff
@@ -256,182 +257,179 @@ private Command prepareListPriority(String args) {
             }
-            if (!(parts.equals(DESC) || parts.equals(ST) || parts.equals(ET) || parts.equals(SD) || parts.equals(ED)
-                    || parts.equals(P)) && lastChar == false) {
-                return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
+        } else if (parameters.equals(END_TIME)) {
+            if (canUpdateEndTime(newValue)) {
+                return true;
+            } else {
+                return false;
             }
+        } else {// edit priority
+            if (canUpdatePriority(newValue)) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+    }
 
+    private boolean canUpdateStartDate(String newValue) {
+        taskComponentArray[Task.START_DATE] = newValue;
+        if (TaskDate.isValidDate(taskComponentArray[Task.START_DATE])) {
+            hasTaskComponentArray[Task.START_DATE_COMPONENT] = true;
+            return true;
+        } else {
+            return false;
         }
-        
-        try {
-            return new EditCommand(targetIdx, new Description(description), hasChangedDescription,
-                    new TaskDate(startDate), hasChangedStartDate, new TaskDate(endDate), hasChangedEndDate,
-                    new TaskTime(startTime), hasChangedStartTime, new TaskTime(endTime), hasChangedEndTime,
-                    new TaskPriority(taskPriority), hasChangedPriority);
-        } catch (IllegalValueException ive) {
-            return new IncorrectCommand(ive.getMessage());
+    }
+
+    private boolean canUpdateEndDate(String newValue) {
+        taskComponentArray[Task.END_DATE] = newValue;
+        if (TaskDate.isValidDate(taskComponentArray[Task.END_DATE])) {
+            hasTaskComponentArray[Task.END_DATE_COMPONENT] = true;
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private boolean canUpdateStartTime(String newValue) {
+        taskComponentArray[Task.START_TIME] = newValue;
+        if (TaskTime.isValidTime(taskComponentArray[Task.START_TIME])) {
+            hasTaskComponentArray[Task.START_TIME_COMPONENT] = true;
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private boolean canUpdateEndTime(String newValue) {
+        taskComponentArray[Task.END_TIME] = newValue;
+        if (TaskTime.isValidTime(taskComponentArray[Task.END_TIME])) {
+            hasTaskComponentArray[Task.END_TIME_COMPONENT] = true;
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private boolean canUpdatePriority(String newValue) {
+        taskComponentArray[Task.TASK_PRIORITY] = newValue;
+        if (TaskPriority.isValidPriority(taskComponentArray[Task.TASK_PRIORITY])) {
+            hasTaskComponentArray[Task.PRIORITY_COMPONENT] = true;
+            return true;
+        } else {
+            return false;
         }
+    }
```

Comment:
These 4 methods seem to be very similar. Perhaps the similar code can be extracted into a more general method?